modules/oe_theme_content_event/tests/src/FunctionalJavascript/LivestreamDisplayDisclosingTest.php

```diff
@@ -0,0 +1,140 @@
+          'value' => $start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+          'end_value' => $start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+        ],
+        'oe_event_languages' => [
+          ['target_id' => 'http://publications.europa.eu/resource/authority/language/EST'],
+          ['target_id' => 'http://publications.europa.eu/resource/authority/language/FRA'],
+        ],
+        'oe_author' => 'http://publications.europa.eu/resource/authority/corporate-body/ACJHR',
+        'oe_content_content_owner' => 'http://publications.europa.eu/resource/authority/corporate-body/COMMU',
+        'uid' => 0,
+        'status' => 1,
+      ]);
+    $node->save();
+    $this->drupalGet($node->toUrl());
+    $livetime_elements = $this->getSession()->getPage()->findAll('css', '[data-livestream-element]');
+    $this->assertCount(0, $livetime_elements);
+    $livestream_js = $this->xpath("//script[contains(@src, 'js/event_livestream.js')]");
+    $this->assertCount(0, $livestream_js);
+
+    // Set livestream start date in 4 hours later.
+    $start_date = (clone $static_time)->modify('+4 hours');
+    $node->set('oe_event_online_type', 'livestream');
+    $node->set('oe_event_online_link', [
+      'uri' => 'http://www.example.com/online_link',
+      'title' => 'Link to online event',
+    ]);
+    $node->set('oe_event_online_description', 'Online event description');
+    $node->set('oe_event_online_dates', [
+      'value' => $start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+      'end_value' => $start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+    ]);
+    $node->save();
+    $this->drupalGet($node->toUrl());
+    $livestream_js = $this->xpath("//script[contains(@src, 'js/event_livestream.js')]");
+    $this->assertCount(1, $livestream_js);
+    $livetime_elements = $this->getSession()->getPage()->findAll('css', '[data-livestream-element]');
+    $this->assertCount(2, $livetime_elements);
+    foreach ($livetime_elements as $livetime_element) {
+      $this->assertFalse($livetime_element->isVisible());
+    }
+
+    // Set livestream start date in 20 seconds later.
+    $start_date = (clone $static_time)->modify('+20 seconds');
+    $node->set('oe_event_online_dates', [
+      'value' => $start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+      'end_value' => $start_date->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT),
+    ]);
+    $node->save();
+    $this->drupalGet($node->toUrl());
+    $livestream_js = $this->xpath("//script[contains(@src, 'js/event_livestream.js')]");
+    $this->assertCount(1, $livestream_js);
+    $livetime_elements = $this->getSession()->getPage()->findAll('css', '[data-livestream-element]');
+    $this->assertCount(2, $livetime_elements);
+    $this->getSession()->wait(20000);
+    foreach ($livetime_elements as $livetime_element) {
+      $this->assertTrue($livetime_element->isVisible());
```

Comment:
We should assert that the time is visible during and after the livestream ended.